src/xml_output.rs

```diff
@@ -7,45 +7,63 @@ use quick_xml::events::{BytesDecl, BytesEnd, BytesStart, BytesText, Event};
 use quick_xml::Writer;
 
 use crate::binxml::value_variant::BinXmlValue;
+use crate::ParserSettings;
 use failure::{bail, format_err, Error};
-use std::borrow::{Cow};
+use std::borrow::Cow;
+
 
 pub trait BinXmlOutput<W: Write> {
     /// Implementors are expected to provide a `std::Write` target.
     /// The record will be written to the target.
-    fn with_writer(target: W) -> Self;
+    fn with_writer(target: W, settings: &ParserSettings) -> Self;
 
     /// Consumes the output, returning control of the inner writer to the caller.
-    fn into_writer(self) -> Result<W, Error>;
+    fn into_writer(self, settings: &ParserSettings) -> Result<W, Error>;
```

Comment:
It's kind of ugly to pass the settings over and over again.
It also doesn't really make any sense to use two different settings in the same Output instance.
Maybe only pass it to `with_writer`, and let the parser copy anything it needs into it's internal state.